build/Jenkinsfile.pr

```diff
@@ -0,0 +1,261 @@
+#!/usr/bin/env groovy
+
+@Library('shared-pipelines') _
+import org.mattermost.Utils
+import java.util.*;
+def utils = new org.mattermost.Utils()
+def rnd = UUID.randomUUID().toString()
+
+pipeline {
+    agent {
+        label 'default-mm-builder'
+    }
+
+    options { buildDiscarder(logRotator(numToKeepStr: '3')) }
+
+    environment {
+        DCNAME = "${rnd}"
+    }
+
+    stages {
+        stage('Stop Old Builds') {
+            steps {
+                script {
+                    utils.stopOldBuilds()
+                }
+            }
+        }
+
+        stage('Checkout'){
+            steps {
+                cleanWs notFailBuild: true
+                sh """
+                    mkdir -p src/github.com/mattermost/mattermost-server
+                    mkdir -p src/github.com/mattermost/mattermost-webapp
+                    mkdir -p src/github.com/mattermost/enterprise
+                """
+                dir('src/github.com/mattermost/mattermost-server') {
+                    checkout scm
```

Comment:
Should we use the same configuration as blow for this too?

Reply:
here is better use the scm, jenkins will retrieve the correct branch/PR for us